packages/grid/src/vaadin-grid-selection-column.js

```diff
@@ -259,7 +259,7 @@ class GridSelectionColumn extends GridColumn {
   /** @private */
   __onSelectedItemsChanged() {
     this._selectAllChangeLock = true;
-    if (Array.isArray(this._grid.items)) {
+    if (Array.isArray(this._grid.items) && this._grid.dataProvider) {
```

Comment:
Would it make sense to extract this check to a separate method so that we can reuse it also here?

https://github.com/vaadin/web-components/blob/9e154f4e98a5651ffc65d72ca5855aaae54675ba/packages/grid/src/vaadin-grid-selection-column.js#L187-L188

Reply:
Makes sense. Done.